### ISSUE TITLE
fails if undefined value is passed

### ISSUE THREAD

**Issue by yoshuawuyts** [substack/hyperx#8]
The following code returns an error:
```js
var foo = undefined
hx`<h1>${foo}</h1>`
```
```txt
Uncaught TypeError: Cannot read property '0' of undefined
```

I think there are two ways of resolving this:
1. `null` and `undefined` values are treated as blank (e.g. yield `<h1></h1>`)
2. `null` and `undefined` values are treated as errors that should never be rendered, and an appropriate error message is thrown

I think my preference goes out to option 1, as visual quirks in UI are generally non-critical; though perhaps there are good arguments for option 2 which is probably more correct.

**Comment by almost** [comment]
I'd definitely vote for 1, that allows much more natural code in some places

**Comment by substack** [comment]
What about turning undefined and null values into the strings `undefined` and `null`?

**Comment by almost** [comment]
That to me seems much less useful. It's a very rare thing that I'd actually want the words "null" or "undefined" in my html output and in those cases I think I'd rather be explicit. But a function that maybe returns some markup or maybe returns nothing? I do that all the time!

**Comment by substack** [comment]
Merged this suggestion into 1.1.0: now `null` and `undefined` result in an empty string. Other types get coerced to strings to avoid type errors later on.
